CrewChiefV4/CrewChief.cs

```diff
@@ -408,11 +413,24 @@ public Boolean Run(String filenameToRun, int interval, Boolean dumpToFile)
                     nextRunTime.Add(TimeSpan.FromMilliseconds(updateTweak));
                     if (!loadDataFromFile)
                     {
-                        if (gameDefinition.processName == null || Utilities.IsGameRunning(gameDefinition.processName))
+                        // Turns our checking for running process by name is an expensive system call.  So don't do that on every tick.
+                        if (now > nextProcessStateCheck)
+                        {
+                            nextProcessStateCheck = now.Add(
+                                TimeSpan.FromMilliseconds(isGameProcessRunning ? timeBetweenProcDisconnectCheckMillis : timeBetweenProcConnectCheckMillis));
+                            isGameProcessRunning = Utilities.IsGameRunning(gameDefinition.processName);
+                        }
+
+                        if (gameDefinition.processName == null || isGameProcessRunning)
                         {
                             if (!mapped)
                             {
                                 mapped = gameDataReader.Initialise();
+
```

Comment:
Jim, if you do not like it I will remove this.  But what's the point of frequent retries throwing exceptions around if resources aren't ready?